### PR TITLE
Added a block to trigger when switch status changes

### PR DESCRIPTION
**Pull request by ghost**
Added a block to trigger when switch status changes

### CODE REVIEW

File: PAYFormBuilder/controller/builder/PAYFormTableBuilder.h [hunk 1]
```diff
@@ -107,7 +107,6 @@
  *                      A SectionBuilder is passed to accomplish this.
  */
 - (void)addSectionWithName:(NSString *)name labelStyle:(PAYFormTableLabelStyle)style contentBlock:(void(^)(PAYFormSectionBuilder *))contentBlock;
-
```
Comment (Eldorado234) [inline PAYFormBuilder/controller/builder/PAYFormTableBuilder.h]: Please revert the whitespace change

File: PAYFormBuilder/controller/builder/PAYFormSectionBuilder.h [hunk 2]
```diff
@@ -242,6 +242,8 @@
  */
 - (PAYFormSwitch *)addSwitchWithName:(NSString *)name configureBlock:(void(^)(PAYFormSwitch *))configureBlock;
 
+- (PAYFormSwitch *)addSwitchWithName:(NSString *)name configureBlock:(void(^)(PAYFormSwitch *))configureBlock changeBlock:(PAYFormSwitchChangeBlock)changeBlock;
```
Comment (Eldorado234) [inline PAYFormBuilder/controller/builder/PAYFormSectionBuilder.h]: Please add the missing documentation

File: PAYFormBuilder/controller/builder/PAYFormSectionBuilder.h [hunk 3]
```diff
@@ -9,11 +9,11 @@
 #import <Foundation/Foundation.h>
 #import "PAYFormButton.h"
 #import "PAYFormButtonGroupBuilder.h"
+#import "PAYFormSwitch.h"
 
 @class PAYFormSingleLineTextField;
 @class PAYFormMultiLineTextField;
 @class PAYFormButton;
-@class PAYFormSwitch;
```
Comment (Eldorado234) [inline PAYFormBuilder/controller/builder/PAYFormSectionBuilder.h]: Why did you change this?

File: PAYFormBuilder/controller/ViewHolder/PAYFormSwitch.m [hunk 4]
```diff
@@ -46,8 +46,21 @@ - (void)setEnabled:(BOOL)enabled {
     self.switchControl.enabled = enabled;
 }
 
+- (void)setChangeBlock:(PAYFormSwitchChangeBlock)changeBlock {
+    [self.switchControl addTarget:self
```
Comment (Eldorado234) [inline PAYFormBuilder/controller/ViewHolder/PAYFormSwitch.m]: The action will be added multiple times when the changeBlock is assigned

File: PAYFormBuilder/controller/builder/PAYFormSectionBuilder.h (hunk 3 above)
Comment (anandrahul) [inline PAYFormBuilder/controller/builder/PAYFormSectionBuilder.h]: Change block won't be visible otherwise

File: PAYFormBuilder/controller/ViewHolder/PAYFormSwitch.h [hunk 5]
```diff
@@ -9,6 +9,10 @@
 #import "PAYFormView.h"
 #import "PAYValidatableFormCell.h"
 
+@class PAYFormSwitch;
```
Comment (Eldorado234) [inline PAYFormBuilder/controller/ViewHolder/PAYFormSwitch.h]: Probably not needed?

Comment (anandrahul) [inline PAYFormBuilder/controller/ViewHolder/PAYFormSwitch.h]: I was not compiling initially. removed.

File: PAYFormBuilder/controller/ViewHolder/PAYFormSwitch.m (hunk 4 above)
Comment (anandrahul) [inline PAYFormBuilder/controller/ViewHolder/PAYFormSwitch.m]: Tested, gets called only once.

**Comment by Eldorado234** [comment]
Thanks for the PR

**Comment by anandrahul** [comment]
You're welcome 👍
